plugins/modules/custom_images.py

```diff
@@ -0,0 +1,315 @@
+  returned: success
+  type: dict
+  sample: {
+    "md5": "5b3a1f21cde154cfb522b582f44f1a87",
+    "sha256": "5b03bcbd00b687e08791694e47d235a487c294e58ca3b1af704120123aa3f4e6"
+  }
+user_data_handling:
+  description: How user_data will be handled when creating a server. There are
+    currently two options, "pass-through" and "extend-cloud-config".
+  returned: success
+  type: str
+  sample: "pass-through"
+tags:
+  description: Tags assosiated with the custom image.
+  returned: success
+  type: dict
+  sample: { 'project': 'my project' }
+status:
+  description: Shows the progress of an import. Values are one of
+    "in_progress", "success" or "error".
+  returned: success
+  type: str
+  sample: "in_progress"
+state:
+  description: The current status of the custom image.
+  returned: success
+  type: str
+  sample: present
+'''
+
+from ansible.module_utils.basic import (
+    AnsibleModule,
+)
+from ..module_utils.api import (
+    AnsibleCloudscaleBase,
+    cloudscale_argument_spec,
+)
+
+
+class AnsibleCloudscaleCustomImages(AnsibleCloudscaleBase):
+
+    def query(self):
+
+        # Initialize
+        self._resource_data = self.init_resource()
+
+        # Set import status
+        importstatus = False
+        if 'importstatus' in self._module.params:
+            importstatus = self._module.params['importstatus']
+            del self._module.params['importstatus']
+
+        # Query by UUID
+        uuid = self._module.params[self.resource_key_uuid]
+
+        if uuid is not None:
+
+            # Get image by uuid
+            resource = self._get('%s/%s' % (self.resource_name, uuid))
+
+            updated = False
+            if resource:
+                # Check if slug, tags or user_data_handling needs update
+                for key in ['slug', 'tags', 'user_data_handling']:
+                    if self._module.params[key] is not None:
+                        if self.find_difference(key,
+                                                resource,
+                                                self._module.params[key]):
+                            updated = True
+
+            # Get image import status by uuid
+            if not updated and not self._module.params['state'] == 'absent':
+                if importstatus is True:
+                    self.resource_name += '/import'
+                resource = self._get('%s/%s' % (self.resource_name, uuid))
+            if resource:
+                self._resource_data = resource
+                self._resource_data['state'] = 'present'
+        else:
+            # Get all images
+            resources = self._get('%s' % (self.resource_name))
+            # Remove images with other names from result
+            if self._module.params['name'] is not None:
+                key = 'name'
+            elif self._module.params['slug'] is not None:
+                key = 'slug'
+            elif self._module.params['tags'] is not None:
```

Comment:
I would add an `else` clause, that raises an error if none of the other statements are True.

Reply:
The module requires one of the parameters `name`, `uuid`, `slug` or `tags`: `required_one_of=(('name', 'uuid', 'slug', 'tags'),)`.

The `uuid` case is handled before, so the `else` clause would be dead code.